### ISSUE TITLE
Pageable Always Adding From Entity to Sort Field Name 

### ISSUE THREAD

**Issue by krankkkk** [spring-projects/spring-data-jpa#2960]
Regression of #1066

Error now reoccurs on Spring Boot 3.1

 When using `@Query` in a JPA repository along with `Pageable` and attempting to sort on a field in a joined entity, the from entity gets prepended to the sort field. For example, if I do the following query:

```java
@Query(value = "select p from Customer c join c.productOrder p where p.delayed = true")
```

 If I pass a `PageRequest` sorting on a field in the from entity (`Customer`), everything will work fine:

```java
PageRequest page = new PageRequest(0, 20, Direction.DESC, "lastName");
 ```

 However, if I simply decide to sort by a field in the `ProductOrder` entity with this `PageRequest`:

 ```java
PageRequest page2 = new PageRequest(0, 20, Direction.DESC, "p.lineItems");
 ```
 
 I will get the following error:
 
 ```
 org.springframework.dao.InvalidDataAccessApiUsageException: org.hibernate.query.sqm.InterpretationException: Error interpreting query [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]; this may indicate a semantic (user query) problem or a bug in the parser [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]
	<snip>
Caused by: java.lang.IllegalArgumentException: org.hibernate.query.sqm.InterpretationException: Error interpreting query [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]; this may indicate a semantic (user query) problem or a bug in the parser [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]
	<snip>
Caused by: org.hibernate.query.sqm.InterpretationException: Error interpreting query [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]; this may indicate a semantic (user query) problem or a bug in the parser [select p from Customer c join c.productOrder p where p.delayed = true order by c.p.lineItems desc]
	<snip>
Caused by: java.lang.IllegalArgumentException: org.hibernate.query.SemanticException: Could not resolve attribute 'p' of 'hello.Customer'
	<snip>
Caused by: org.hibernate.query.SemanticException: Could not resolve attribute 'p' of 'hello.Customer'
	<snip>
 ```
`c.` was prepended onto the sort field.

 Updated reproducer project: [Archive.zip](https://github.com/spring-projects/spring-data-jpa/files/11517715/Archive.zip)

**Comment by gregturn** [comment]
Looks like this issue not only regresses against #1066, but touches on #664, an older ticket that may have been the origin of handling JOINs with aliases in Sorts.

**Comment by gregturn** [comment]
This is certainly impacted by #3002.

**Comment by gregturn** [comment]
Based on https://github.com/spring-projects/spring-data-jpa/pull/3010#issuecomment-1586722460, #3002 is no longer relevant.

**Comment by gregturn** [comment]
Merged to `main` and backported to `3.1.x`.
